sotodlib/coords/pointing_model.py

```diff
@@ -141,52 +145,82 @@ def model_lat_v1(params, az, el, roll):
-      of the corotator.
-    - cr_offset: The corotator encoder offset.
-    - el_{xi, eta}_offset: The offset between the elevation exis and the
-      corotator axis.
-    - mir_{xi, eta}_offset: The offset between the mirror's axis and the elevation
-      axis (ie: a tilt in the mirrors).
-    - az_offset: The azimuth encoder offset.
-    - el_offset: The elevation encoder offset.
     """
-    _params = {
-        "az_offset": 0,
-        "el_offset": 0,
-        "cr_offset": 0,
-        "el_xi_offset": 0,
-        "el_eta_offset": 0,
-        "rx_xi_offset": 0,
-        "rx_eta_offset": 0,
-        "mir_xi_offset": 0,
-        "mir_eta_offset": 0,
-    }
-    _params = _params.copy()
-    _params.update(params)
-    params = _params
-    cr = el - roll - np.deg2rad(60)
-    for key, val in params.items():
-        if not isinstance(val, (float, int)):
+    _p = dict(param_defaults['lat_v1'])
+    if isinstance(params, dict):
+        _p.update(params)
+    else:
+        _p.update({k: params[k] for k in params._fields.keys()})
+    params, _p = _p, None   
+         
+    for k, v in params.items():
+        if k == 'version':
             continue
-        params[key] = np.deg2rad(val)
-    q_enc = quat.rotation_lonlat(
-        -1 * (az.copy() + params["az_offset"]), el.copy() + params["el_offset"]
+        if k not in param_defaults['lat_v1'] and v != 0.:
+            raise ValueError(f'Handling of model param "{k}" is not implemented.')
+    
+    # Here we reconstruct the naive corotator value before applying 
+    # the elevation encoder offset.
+    cr = el - roll - np.deg2rad(60)    
+    
+    az_orig = az.copy()
+    az = az + params['enc_offset_az']
+    el = el + params['enc_offset_el'] 
+    cr = cr + params['enc_offset_cr']
+    
+    # Lonlat rotation with az and el encoder offsets included.
+    q_lonlat = quat.rotation_lonlat(-1 * az, el)
+
+    # Rotation that takes a vector in elevation-hub centered 
+    # coordinates to final boresight centered coordinates. 
+    # Accounts for any mirror-related offsets.
+    q_mir_center = ~quat.rotation_xieta(
+        params['mir_center_xi0'],
+        params['mir_center_eta0']
     )
-    q_mir = quat.rotation_xieta(params["mir_xi_offset"], params["mir_eta_offset"])
-    q_el_roll = quat.euler(2, el.copy() + params["el_offset"] - np.deg2rad(60))
-    q_tel = quat.rotation_xieta(params["el_xi_offset"], params["el_eta_offset"])
-    q_cr_roll = quat.euler(2, -1 * cr - params["cr_offset"])
-    q_rx = quat.rotation_xieta(params["rx_xi_offset"], params["rx_eta_offset"])
-    new_az, el, roll = (
-        quat.decompose_lonlat(q_enc * q_mir * q_el_roll * q_tel * q_cr_roll * q_rx)
-        * np.array([-1, 1, 1])[..., None]
+
+    # Elevation component of roll motion
+    q_el_roll = quat.euler(2, el - np.deg2rad(60))
+    
+    # Rotation that takes a vector in telescope's elevation-hub centered 
+    # coordinates to mirror-centered coordinates
```

Comment:
I think this should be "takes a vector in telescope's corotator-centered coordinates to el-hub-centered coordinates"